### PR TITLE
Ceds 3570 payload requirement for additional info

### PR DESCRIPTION
**Pull request by darrylmurphy**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala [hunk 1]
```diff
@@ -43,14 +43,29 @@ class AdditionalInformationBuilder @Inject()() extends ModifyingBuilder[ExportIt
...
-        wcoGovernmentAgencyGoodsItem.getAdditionalInformation.add(buildAdditionalInformation(additionalInformation))
+  def buildThenAdd(
+    exportItem: ExportItem,
+    declarantIsExporter: Option[DeclarantIsExporter],
+    wcoGovernmentAgencyGoodsItem: GoodsShipment.GovernmentAgencyGoodsItem
+  ): Unit = {
+
+    val additionalInformation = AdditionalInformation(code = "00400", description = "EXPORTER")
 
+    lazy val infoEnteredByUser = exportItem.additionalInformation.exists { additionalInformationData =>
```
Comment (udinobi) [inline app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala:54]: thb this block is a bit convoluted.

Comment (glidester) [reply app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala]: Agreed, could be simplified to something like:
```
lazy val infoEnteredByUser = exportItem.additionalInformation
      .map{ _.items.exists( _ == `additionalInformation`) }
      .getOrElse(false)
```

Comment (glidester) [reply app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala]: Any reason it is defined as lazy?

File: app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala [hunk 2]
```diff
@@ -43,14 +43,25 @@ class AdditionalInformationBuilder @Inject()() extends ModifyingBuilder[ExportIt
...
+    exportItem: ExportItem,
+    declarantIsExporter: Option[DeclarantIsExporter],
+    wcoGovernmentAgencyGoodsItem: GoodsShipment.GovernmentAgencyGoodsItem
+  ): Unit = {
+
+    val additionalInformation = AdditionalInformation(code = "00400", description = "EXPORTER")
 
+    lazy val infoEnteredByUser = exportItem.additionalInformation.exists { additionalInformationData => {
+        additionalInformationData.items.contains(additionalInformation)
+      }
```
Comment (udinobi) [inline app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala:56]: Really nit: the open-close braces add noise and could be removed

**Review by udinobi — APPROVED** [review]

**Review by udinobi — APPROVED** [review]

**Pull request merged**